src/coordinator/ts/m3db/convert_test.go

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package m3db
+
+import (
+	"fmt"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/m3db/m3db/src/dbnode/encoding"
+	"github.com/m3db/m3db/src/dbnode/encoding/m3tsz"
+	"github.com/m3db/m3db/src/dbnode/ts"
+	"github.com/m3db/m3db/src/dbnode/x/xio"
+	"github.com/m3db/m3x/checked"
+	"github.com/m3db/m3x/ident"
+	xtime "github.com/m3db/m3x/time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestConversion(t *testing.T) {
+	iterator := BuildTestSeriesIterator(t)
+	iterators := encoding.NewSeriesIterators([]encoding.SeriesIterator{iterator}, nil)
+	blocks, err := convertM3DBSeriesIterators(iterators)
+	require.NoError(t, err)
+
+	for _, block := range blocks {
+		assert.Equal(t, "foo", block.ID.String())
+		assert.Equal(t, "namespace", block.Namespace.String())
+
+		blockOne := block.Blocks[0].SeriesIterator
+		blockTwo := block.Blocks[1].SeriesIterator
+
+		start = time.Now().Truncate(time.Hour).Add(2 * time.Minute)
+		middle = start.Add(30 * time.Minute)
+		end = middle.Add(30 * time.Minute)
+
+		assert.Equal(t, start, blockOne.Start())
+		// assert.Equal(t, middle, blockOne.End())
+
+		for i := 1; blockOne.Next(); i++ {
+			dp, _, _ := blockOne.Current()
+			assert.Equal(t, float64(i), dp.Value)
```

Comment:
Check the data point timestamp too? Also this will have to change when the other PR drops since [1,2] should not appear

Reply:
You think we need to check the timestamp of each datapoint even if we verify the block start and end?